### DISCUSSION TITLE
Offerbook UX

### DISCUSSION THREAD

**Discussion by chimp1984** [bisq-network/bisq#5566] [General]
As we want to have only one global offerbook (not buy/sell side) we cannot use the ask or bid side for the table colums as there is none if the user has diabled the currency filter. Instead we should use the base currency/quote currency distinction.
We also support altcoin-altcoin and fiat-fiat exchange so its gets also more tricky to define whats the base currency.

The base currency is the one on the left side of the price quote [1]. The quote currency the one on the right side.
E.g. BTC/USD means BTC is the base currency and USD the quote currency. You get the amount stated in the quote for 1 unit of the base currency (BTC), e.g. 50 000 USD for 1 BTC. The confusing thing is that they use the slash and people interpret it as division and then it would be the other way round (50000 USD/BTC)...

So if we use the base currency and quote currency as table columns we can support any currency pair also EUR/USD and USD/EUR - the users will define whats the base currency for those pairs where it is not a clear consensus.

I am not sure about the best term to use as `base currency` and `quote currency` might be not that well known.
Having base and quote currencies we can use the layout used in the create offer as well: 
`base currency x price = quote currency` so the table columns can be:
`| base currency | price | quote currency |`

I am aware that the ask and bid would be more user-centric but as it does not cover the case when no currency for ask/bid are selected I think its not the right model.

[1] https://www.investopedia.com/terms/b/basecurrency.asp

**Comment by eyalron33** [comment]
I already got confused while reading it and have to translate in my brain each time that I see those things, so I can't imagine it will be easier for users...

So a different question, because I feel I missed a step in the discussion. Why do we want "one global offerbook"? What's the goal of having this instead of/in addition of an offer book for each pair?

**Reply by chimp1984** [reply]
We want to support any trade types like EUR to USD or GRIN-BEAM. In those cases it is not clear what is the base curreny and how to quote the price. Is it EUR/USD or USD/EUR? Answer is both are valid and the user has to define it then. But all that makes the current concept with BUY and SELL invalid as there is no "money" for a "commodity" situation anymore as it is when you exchange fiat to BTC or altcoin to BTC as in Bisq now.
Usually the dominant currency is considered the quote currency - that what you consider the money side. Fiat in case of BTC/Fiat and BTC in case of Altcoin/BTC. But when yu trade fiat/fiat or altcoin/altcoin it becomes less clear.

**Reply by eyalron33** [reply]
Yes, I know that, but maybe I don't understand what you mean by "one global offerbook"?

LIke, I understand "one global offerbook" as a screen with all offers of all possible pairs. Do you mean simply an offerbook for one pair where it's unclear what is the base one?

If yes, isn't it the discussion we [had here](https://github.com/bisq-network/projects/issues/57)? Then what's wrong with " one-column Offerbook (so all offers in the offer book buy and sell the same asset)" discussed there? You don't need to write COIN1/COIN2, simply write a screen title of "offers for selling COIN1 for COIN2" which is super clear.

Does sticking to a notation of COIN1-COIN2 or COIN1/COIN2 brings anything?

**Reply by chimp1984** [reply]
I think its a misunderstanding. One global offerbook and what you refer to one-column Offerbook is the same (though its not one colums as the colums are amounr, price, details,....).
So in that model the user can select any asset on ask and bid side as well select "any/show all". In that case what should be the column names for the 2 currencies? Price quoteation for fiat and altcoin is inverse (BTC/USD and XMR/BTC) - that makes it not easier when you have all currencies mixed.

**Reply by chimp1984** [reply]
Actually I think its ok to use the ask and bid side instead the base and quote amounts. Its more client centric then.

**Reply by eyalron33** [reply]
Btw, for me personally ask/offer was always a bit more clear than ask/bid (but it's personal taste)

**Comment by cd2357** [comment]
One solution is to express the exchange rate as BTCUSD instead of BTC/USD, which implies division.

Its pretty intuitive to equate "BTCUSD" to "1 BTC expressed in USD".

**Reply by chimp1984** [reply]
> One solution is to express the exchange rate as BTCUSD instead of BTC/USD, which implies division.
> 
> Its pretty intuitive to equate "BTCUSD" to "1 BTC expressed in USD".

Yes, might be an option. It is one common way how to display it. I think a "_" or  "-" is also used sometimes. Without separator it might be harder to read when it comes to more exotic currency pairs (we should be able to support also altcoin-altcoin - there it could lead to ambiguity in some cases ).
